File: services/event_service/app/database/models/status_event.py
from sqlalchemy import Column, ForeignKey, Integer

from database.schemas.status import StatusEventSchema

from . import base


class StatusEvent(base.BaseModel):
    __tablename__ = "status_event"

    event_id = Column(
        Integer, ForeignKey("event.id", ondelete="CASCADE"), nullable=False
    )
    status_id = Column(
        Integer, ForeignKey("status.id", ondelete="CASCADE"), nullable=False
    )

    def to_read_model(self) -> StatusEventSchema:
        return StatusEventSchema(
            id=self.id,
            event_id=self.event_id,
            status_id=self.status_id,
        )

    @classmethod
    def convert_scheme(cls):
        return StatusEventSchema

File: services/event_service_golang/internal/models/status_event.go
package models

type StatusEvent struct {
	tableName struct{} `pg:"status_event"`
	EventID   int      `pg:"event_id,pk"`
	Event     *Event   `pg:"rel:has-one"`

	StatusID int     `pg:"status_id,pk"`
	Status   *Status `pg:"rel:has-one"`
}
